Diff/Diff.sublime-syntax

```diff
@@ -1,75 +1,381 @@
+  | Prereq:[ ]
+  | ---\ [^%]
+  | \*\*\*.*\d{4}\s*$
+  | \d+(,\d+)* [adc] \d+(,\d+)* $
+  )
+
+variables:
+  hash64: '[0-9a-fA-F]{64}'
+  hash40: '[0-9a-fA-F]{40}'
+  hash_full: (?:{{hash40}}|{{hash64}})
+  git_first_line: |-
+    ^(?x:
+      From [ ] \w+@z [ ]
+      Thu [ ] Jan [ ]{2} 1 [ ] 00:00:00 [ ] 1970  # git-send-email
+    | From [ ] {{hash_full}} [ ]
+      Mon [ ] Sep [ ]   17 [ ] 00:00:00 [ ] 2001  # git-format-patch
+    )
+  eol: (?:$\r?\n?)
 
-scope: source.diff
 contexts:
+
+###[ PROTOTYPES ]##############################################################
+
+  pop-immediately:
+    - match: ''
+      pop: 1
+
+  pop-eol:
+    - match: '{{eol}}'
+      pop: 1
+
+  pop-before-eol:
+    - match: (?={{eol}})
+      pop: 1
+
+  line-ranges:
+    - include: line-ranges-normal
+    - include: line-ranges-context
+    - include: line-ranges-unified
+
+  deltas:
+    - include: deltas-normal
+    - include: deltas-context
+    - include: deltas-unified
+
+  unified:
+    - include: headers-unified
+    - include: line-ranges-unified
+    - include: deltas-unified
+
+  diff3:
+    - include: diff3-normal
+    - include: diff3-edit
+
   main:
-    - match: '^((\*{15})|(={67})|(-{3}))$\n?'
+    - match: ^(?={{git_first_line}})
+      push: Packages/Git Formats/Git Diff.sublime-syntax#email-first-line
+    - include: leading-text
+    - include: line-ranges
+    - include: headers
+    - include: deltas
+    - include: diff3-normal
+
+###[ LEADING TEXT ]############################################################
+
+  leading-text:
+    - match: ^(Index)(:)[ \t]*
+      captures:
+        1: meta.mapping.key.diff keyword.other.diff
+        2: punctuation.separator.key-value.diff
+      push:
+        - meta_scope: meta.mapping.diff meta.diff.index meta.index.diff
+        - meta_content_scope: meta.mapping.value.diff meta.toc-list.file-name.diff
+        - include: pop-eol
+        - include: path-separators
+    - match: ^(Prereq)(:)[ \t]*
+      captures:
+        1: meta.mapping.key.diff keyword.other.diff
+        2: punctuation.separator.key-value.diff
+      push:
+        - meta_scope: meta.mapping.diff
+        - meta_content_scope: meta.mapping.value.diff string.unquoted.diff
+        - include: pop-eol
+        - include: path-separators
+
+###[ HEADERS ]#################################################################
+
+  headers:
+    - include: headers-context
+    - include: headers-unified
+    - match: ^(={4}) .+(?= - )
+      scope: meta.diff.header.from-file meta.header.from-file.diff
+      captures:
+        1: punctuation.definition.from-file.diff
+      push:
+        - match: '( (-) )(.* (={4})){{eol}}'
+          captures:
+            1: meta.header.diff
+            2: punctuation.separator.sequence.diff
+            3: meta.diff.header.to-file meta.header.to-file.diff
+            4: punctuation.definition.to-file.diff
+          pop: 1
+        - include: pop-immediately
+    - match: ^(\*{15}|={67}){{eol}}
```

Comment:
Also, does anyone have any idea what these rules are for? I left them in there (refactored), but I cannot find any citation, or even an example outside the test file for these specific numbers of `=` or `*` being a separator.

Reply:
The `***************` appears to be a separator between hunks in a file in "Context" mode. Still curious about the 67 consecutive `=`s.

Reply:
Maybe just a similar app specific separator.